perun-oidc/src/main/java/cz/metacentrum/perun/oidc/OIDC.java

```diff
@@ -70,44 +80,88 @@ public Object process(PerunSession sess, String method, Deserializer params) thr
 	}
 
 
-
-	private Map<String, Object> getUserinfo(PerunSession sess, User user, Collection<String> allowedScopes, Map<String, String> properties)
+	private ObjectNode getUserinfo(final PerunSession sess, final User user, Collection<String> allowedScopes, Map<String, String> properties)
 			throws InternalErrorException, PrivilegeException, UserNotExistsException, WrongAttributeAssignmentException {
 
-		Map<String, Object> userinfo = new HashMap<>();
+		ObjectMapper mapper = new ObjectMapper();
+		ObjectNode userinfo = mapper.createObjectNode();
 
-		for (String scope : allowedScopes) {
+		for (final String scope : allowedScopes) {
 
 			String property = properties.get(scope);
-			if (property == null) {
-				log.warn("No attributes are mapped to scope "+scope+". Configure it in /etc/perun/"+PROPERTIES_FILE);
+			if (property == null || property.isEmpty()) {
+				log.info("No values are mapped to scope "+scope+". Configure them in "+ CONFIG_FILE);
 				continue;
 			}
+			try {
+				JsonNode claims = mapper.readTree(property);
 
-			for (String claim : Arrays.asList(property.split(","))) {
-				if (claim.isEmpty()) {
-					continue;
+				if (!claims.isObject()) {
+					throw new InternalErrorException("Config file "+ CONFIG_FILE +" is wrongly configured. Values have to be valid JSON objects.");
 				}
 
-				try {
-					String claimName = claim.split("/", 2)[0];
-					String attrName = claim.split("/", 2)[1];
+				claims = dfs(sess, user, claims);
+				if (claims != null && claims.isObject()) {
+					userinfo.putAll((ObjectNode) claims);
+				}
+
+			} catch (IOException e) {
+				throw new InternalErrorException("Config file "+ CONFIG_FILE +" is wrongly configured. Values have to be valid JSON objects.", e);
+			}
+
+		}
+
+		return userinfo;
+
+	}
 
-					try {
-						userinfo.put(claimName, sess.getPerun().getAttributesManager().getAttribute(sess, user, attrName).getValue());
 
+	private JsonNode dfs(PerunSession sess, User user, JsonNode node) throws InternalErrorException, WrongAttributeAssignmentException, PrivilegeException, UserNotExistsException {
+
+		ObjectMapper mapper = new ObjectMapper();
+		if (!node.isObject()) {
+
+			if (node.isTextual()) {
+				Matcher m = attrRegex.matcher(node.getTextValue());
+				if (m.matches()) {
+
+					try {
+						Object value = sess.getPerun().getAttributesManager().getAttribute(sess, user, node.getTextValue()).getValue();
+						if (value != null) {
+							return new TextNode(value.toString());
+						} else {
+							log.info("Attribute "+node.getTextValue()+" defined in config "+CONFIG_FILE+" is not present for user "+user);
```

Comment:
I think text does not say problem is there. But attribute is configured there and is not set for the user.